content/intro-to-storybook/react/fr/composite-component.md

```diff
@@ -0,0 +1,282 @@
+---
+title: 'Assembler un composant complexe'
+tocTitle: 'Composant complexe'
+description: 'Assembler un composant complexe à partir de composants plus simples'
+commit: 'c43580d'
+---
+
+Dans le précédent chapitre nous avons construit notre premier composant; ce chapitre prolonge ce que nous avons appris pour construire la TaskList, une liste de tâches. Combinons les composants ensemble et voyons ce qui se passe lorsqu'on introduit plus de complexité.
+
+## Tasklist(Liste des tâches)
+
+Taskbox met l'accent sur les tâches épinglées en les positionnant au-dessus des tâches par défaut. Cela donne deux variantes de la `TaskList` pour laquelle vous devez créer des story: les éléments par défaut et les éléments par défaut et épinglés.
+
+![tâches par défaut et tâches épinglées](/intro-to-storybook/tasklist-states-1.png)
+
+Comme les données de `Task` peuvent être envoyées de manière asynchrone, nous avons **aussi** besoin d'un état de chargement à rendre en l'absence de connexion. De plus, un état vide est nécessaire lorsqu'il n'y a pas de tâches.
+
+![les tâches vides et en cours de chargement](/intro-to-storybook/tasklist-states-2.png)
+
+## Se préparer
+
+Un composant complexe n'est pas très différent des composants de base qu'il contient. Créez un composant `TaskList` et un fichier d'accompagnement : `src/components/TaskList.js` et `src/components/TaskList.stories.js`.
+
+Commencez par une implémentation grossière de la `TaskList`. Vous devrez importer le composant `Task` de la version précédente et lui passer les attributs et les actions en entrée.
+
+```javascript
+// src/components/TaskList.js
+
+import React from 'react';
+
+import Task from './Task';
+
+function TaskList({ loading, tasks, onPinTask, onArchiveTask }) {
+  const events = {
+    onPinTask,
+    onArchiveTask,
+  };
+
+  if (loading) {
+    return <div className="list-items">loading</div>;
+  }
+
+  if (tasks.length === 0) {
+    return <div className="list-items">empty</div>;
+  }
+
+  return (
+    <div className="list-items">
+      {tasks.map(task => (
+        <Task key={task.id} task={task} {...events} />
+      ))}
+    </div>
+  );
+}
+
+export default TaskList;
+```
+
+Ensuite, créez les états de test de la `Tasklist` dans le fichier story.
+
+```javascript
+// src/components/TaskList.stories.js
+
+import React from 'react';
+
+import TaskList from './TaskList';
+import * as TaskStories from './Task.stories';
+
+export default {
+  component: TaskList,
+  title: 'TaskList',
+  decorators: [story => <div style={{ padding: '3rem' }}>{story()}</div>],
+};
+
+const Template = args => <TaskList {...args} />;
+
+export const Default = Template.bind({});
+Default.args = {
+  // Mise en forme des story par la composition d'args.
+  // Les données ont été héritées du story par défaut dans task.stories.js.
```

Comment:
@Draketheb4dass one thing. If you don't mind, leave the comments in the snippets in english. I know that translating them would bring the translation "closer to home" (pardon the bad pun). But general rule of thumb it's best to avoid translate them. 

If you can apply it to all the snippets we'll be good to go and merge this.

Reply:
On it right now.